Map algolia

ethanstrominger opened the pull request:
For environmental vars added, see https://docs.google.com/document/d/1DNklTnhFZWYsJKnOzHK3VhFznAck1UP9biVd9P9wyok/edit

Issues
  - This will. not be protected from development but data is not so sensitive that this should be an issue.  We either need to get algolia to upgrade reach4help "application" (equivalent of an account) which will then let us provide fine grained security or get algolia to give reach4help-dev have same privs as reach4help.
  - If the json includes invalid org type or service type the app may crash.  Logged issue #1426 for this.

Change REACT_APP_ALGOLIA_INDEX_NAME to markers-production


To migrate the data:

- Download json file
- Create a .env file with the four algolia variables and save in map/src/algolia-scripts
cd map/src/algolia-scripts
node import-marker-json-to-algolia.mjs ~/Downloads/markers-prod.json markers-production

reach4help-iam-bot committed:
New translations en.json (Romanian)

reach4help-iam-bot committed:
New translations en.json (Dutch)

reach4help-iam-bot committed:
New translations en.json (Bengali)

reach4help-iam-bot committed:
New translations en.json (Portuguese, Brazilian)

reach4help-iam-bot committed:
New translations en.json (Vietnamese)

reach4help-iam-bot committed:
New translations en.json (Chinese Traditional)

reach4help-iam-bot committed:
New translations en.json (Chinese Simplified)

reach4help-iam-bot committed:
New translations en.json (Ukrainian)

reach4help-iam-bot committed:
New translations en.json (Turkish)

reach4help-iam-bot committed:
New translations en.json (Swedish)

reach4help-iam-bot committed:
New translations en.json (Serbian (Cyrillic))

reach4help-iam-bot committed:
New translations en.json (Russian)

reach4help-iam-bot committed:
New translations en.json (Portuguese)

reach4help-iam-bot committed:
New translations en.json (Norwegian)

reach4help-iam-bot committed:
New translations en.json (Polish)

reach4help-iam-bot committed:
New translations en.json (Korean)

reach4help-iam-bot committed:
New translations en.json (French)

reach4help-iam-bot committed:
New translations en.json (Japanese)

reach4help-iam-bot committed:
New translations en.json (Italian)

reach4help-iam-bot committed:
New translations en.json (Hungarian)

reach4help-iam-bot committed:
New translations en.json (Hebrew)

reach4help-iam-bot committed:
New translations en.json (Finnish)

reach4help-iam-bot committed:
New translations en.json (Greek)

reach4help-iam-bot committed:
New translations en.json (German)

reach4help-iam-bot committed:
New translations en.json (Danish)

reach4help-iam-bot committed:
New translations en.json (Czech)

reach4help-iam-bot committed:
New translations en.json (Catalan)

reach4help-iam-bot committed:
New translations en.json (Arabic)

reach4help-iam-bot committed:
New translations en.json (Afrikaans)

reach4help-iam-bot committed:
New translations en.json (Spanish)

reach4help-iam-bot committed:
New translations en.json (Hindi)

reach4help-iam-bot committed:
New translations en.json (Romanian)

reach4help-iam-bot committed:
New translations en-US.json (Catalan)

reach4help-iam-bot committed:
New translations en-US.json (Arabic)

reach4help-iam-bot committed:
New translations en-US.json (Afrikaans)

reach4help-iam-bot committed:
New translations en-US.json (Spanish)

reach4help-iam-bot committed:
New translations en-US.json (French)

reach4help-iam-bot committed:
New translations en-US.json (Romanian)

reach4help-iam-bot committed:
New translations en-US.json (Danish)

reach4help-iam-bot committed:
New translations en-US.json (Czech)

reach4help-iam-bot committed:
New translations en-US.json (German)

reach4help-iam-bot committed:
New translations en-US.json (Russian)

reach4help-iam-bot committed:
New translations en-US.json (Portuguese, Brazilian)

reach4help-iam-bot committed:
New translations en-US.json (Bengali)

reach4help-iam-bot committed:
New translations en-US.json (Vietnamese)

reach4help-iam-bot committed:
New translations en-US.json (Chinese Traditional)

reach4help-iam-bot committed:
New translations en-US.json (Chinese Simplified)

reach4help-iam-bot committed:
New translations en-US.json (Ukrainian)

reach4help-iam-bot committed:
New translations en-US.json (Turkish)

reach4help-iam-bot committed:
New translations en-US.json (Swedish)

reach4help-iam-bot committed:
New translations en-US.json (Serbian (Cyrillic))

reach4help-iam-bot committed:
New translations en-US.json (Portuguese)

reach4help-iam-bot committed:
New translations en-US.json (Greek)

reach4help-iam-bot committed:
New translations en-US.json (Polish)

reach4help-iam-bot committed:
New translations en-US.json (Norwegian)

reach4help-iam-bot committed:
New translations en-US.json (Dutch)

reach4help-iam-bot committed:
New translations en-US.json (Korean)

reach4help-iam-bot committed:
New translations en-US.json (Japanese)

reach4help-iam-bot committed:
New translations en-US.json (Italian)

reach4help-iam-bot committed:
New translations en-US.json (Hungarian)

reach4help-iam-bot committed:
New translations en-US.json (Hebrew)

reach4help-iam-bot committed:
New translations en-US.json (Finnish)

reach4help-iam-bot committed:
New translations en.json (French)

reach4help-iam-bot committed:
New translations en.json (Hungarian)

reach4help-iam-bot committed:
New translations en.json (Russian)

reach4help-iam-bot committed:
New translations en.json (Portuguese)

reach4help-iam-bot committed:
New translations en.json (Polish)

reach4help-iam-bot committed:
New translations en.json (Norwegian)

reach4help-iam-bot committed:
New translations en.json (Dutch)

reach4help-iam-bot committed:
New translations en.json (Korean)

reach4help-iam-bot committed:
New translations en.json (Japanese)

reach4help-iam-bot committed:
New translations en.json (Italian)

reach4help-iam-bot committed:
New translations en.json (Hebrew)

reach4help-iam-bot committed:
New translations en.json (Swedish)

reach4help-iam-bot committed:
New translations en.json (Finnish)

reach4help-iam-bot committed:
New translations en.json (Greek)

reach4help-iam-bot committed:
New translations en.json (German)

reach4help-iam-bot committed:
New translations en.json (Danish)

reach4help-iam-bot committed:
New translations en.json (Czech)

reach4help-iam-bot committed:
New translations en.json (Catalan)

reach4help-iam-bot committed:
New translations en.json (Arabic)

reach4help-iam-bot committed:
New translations en.json (Afrikaans)

reach4help-iam-bot committed:
New translations en.json (Spanish)

reach4help-iam-bot committed:
New translations en.json (Serbian (Cyrillic))

reach4help-iam-bot committed:
New translations en.json (Turkish)

reach4help-iam-bot committed:
New translations en.json (Ukrainian)

reach4help-iam-bot committed:
New translations en.json (Hindi)

reach4help-iam-bot committed:
New translations en.json (Bengali)

reach4help-iam-bot committed:
New translations en.json (Portuguese, Brazilian)

reach4help-iam-bot committed:
New translations en.json (Vietnamese)

reach4help-iam-bot committed:
New translations en.json (Chinese Traditional)

reach4help-iam-bot committed:
New translations en.json (Chinese Simplified)

reach4help-iam-bot committed:
New translations en-US.json (Hindi)

ethanstrominger committed:
feat: upd yarn.lock for algolia

ethanstrominger committed:
feat: for sharing current state with Ashkay

ethanstrominger committed:
feat: map - debugged scripts for uploading

ethanstrominger committed:
refactor: map - populate algolia scripts

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci] (merge)

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci] (merge)

allcontributors[bot] committed:
docs: update README.md [skip ci] (merge)

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

shayantist committed:
fix: temporarily disable cloud function syncMutualAidWiki stuck in a loop

ethanstrominger committed:
feat: map - minor changes to script

shayantist committed:
Merge pull request #1340 from reach4help/all-contributors/add-Lakshmi-Jayakumar

docs: add Lakshmi-Jayakumar as a contributor

ethanstrominger committed:
feat: replaced firebase fetch w/data fetch

Co-authored-by: Kat Rodriguez <katr400@gmail.com>

ethanstrominger committed:
feat: rename files

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update README.md [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

allcontributors[bot] committed:
docs: update .all-contributorsrc [skip ci]

shayantist committed:
Merge pull request #1342 from reach4help/all-contributors/add-ssameghini

docs: add ssameghini as a contributor

ethanstrominger committed:
feat: map - final touches on Algolia scripts

ethanstrominger committed:
feat: map - minor change to script

ethanstrominger committed:
feat: map - minor change to script

ethanstrominger committed:
Merge branch 'map' into map-algolia

ethanstrominger committed:
feat: map - algolia, wip

ethanstrominger committed:
feat: fixed 20 existing compile errs, renamed firebase.ts, removed firebase from dataDriver.ts

ethanstrominger committed:
feat: map-algolia - try to get add to work

shayantist committed:
fix: pause syncMutualAidHub cloud function until we address duplicates

shayantist committed:
Merge branch 'production' of https://github.com/reach4help/reach4help into staging

shayantist committed:
Merge branch 'staging' of https://github.com/reach4help/reach4help into development

ethanstrominger committed:
refactor: map-algolia, share functions

ethanstrominger committed:
feat: add marker to Algolia from UI

ethanstrominger committed:
feat: model - update MarkerInfo for Algolia

shayantist committed:
feat: update data with newly cleaned data

shayantist committed:
feat: update scripts

ethanstrominger committed:
Merge branch 'map' into development

ethanstrominger committed:
merge readme.md

ethanstrominger committed:
merge readme.md

jensren committed:
Merge branch 'map-algolia' of https://github.com/reach4help/reach4help into map-algolia

jensren committed:
Fixed #1348: Dev Issue: fix compile warnings

Co-authored-by: Ethan Strominger <ethanstrominger2@gmail.com>

jensren committed:
Fix build & lint errors

ethanstrominger committed:
1350 removed spaced from dir name

ethanstrominger committed:
1350 removed spaced from dir name

ethanstrominger committed:
Merge branch 'development' into map-algolia

ethanstrominger committed:
fix: map-algolia:fix yarn lint,mv scripts js to mjs

ethanstrominger committed:
fix: map-algolia:fix yarn lint,mv scripts js to mjs

ethanstrominger committed:
fix: map-algolia:fix yarn lint,mv scripts js to mjs

Co-authored-by: Ethan Strominger <ethanstrominger2@gmail.com>

ethanstrominger committed:
fix: map-algolia:fix yarn lint,mv scripts js to mjs

jensren committed:
fix: map-algolia fix lint errors

jensren committed:
fix: map-algolia fix lint errors

Co-authored-by: Ethan Strominger <ethanstrominger2@gmail.com>

jensren committed:
fix: debugging commit error

jensren committed:
fix: debugging commit error

jensren committed:
fix: debugging commit error

jensren committed:
fix: debugging commit error

ethanstrominger committed:
Merge branch 'map-algolia' into jen/map-algolia

ethanstrominger committed:
Merge pull request #1359 from reach4help/jen/map-algolia

Fix: lint errors

jensren committed:
fix: map-algolia dataDriver can't use mjs

Co-authored-by: Ethan Strominger <ethanstrominger2@gmail.com>

ethanstrominger committed:
Merge pull request #1360 from reach4help/jen/map-algolia/move-file

fix: map-algolia dataDriver can't use mjs

jensren committed:
#1203: add createdAt, updatedAt and verifiedAt

ethanstrominger committed:
fix: fixed script error

ashwinkjoseph committed:
feat: setting up hasura base

shayantist committed:
Merge pull request #1361 from reach4help/jen/map-algolia2/1203-add-createdat

#1203: add createdAt, updatedAt and verifiedAt

ethanstrominger committed:
Merge pull request #1378 from reach4help/map

Map to Map Algolia merge

ethanstrominger committed:
Removed readme.md  merge conflict comments

ethanstrominger committed:
Merge branch 'map' into map-algolia

ethanstrominger committed:
refactor: changed "firebase" to "dataDriver"

ethanstrominger committed:
Merge pull request #1385 from reach4help/map

Map algolia to map merge

ethanstrominger committed:
map: algolia scripts review before push

ethanstrominger committed:
feat: map algolia - add feature

ethanstrominger committed:
fix: map algolia color problem

jensren committed:
refactor: setFilter in App

shayantist commented on map/src/data/dataDriver.ts:
```diff
@@ -2,34 +2,23 @@ import {
-// eslint-disable-next-line import/no-duplicates
-import 'firebase/storage';
-// eslint-disable-next-line import/no-duplicates
-import 'firebase/analytics';
```
Until we integrate Algolia analytics, we shouldn't remove this.

ethanstrominger replied:
This will not affect firebase analytics.    Firebase is no longer used by dataDriver.ts (previously named fireBase.ts)

shayantist commented on map/src/data/dataDriver.ts:
```diff
@@ -147,22 +138,28 @@ export const removeInformationListener = (l: InformationListener) => {
   listeners.delete(l);
 };
 
+// TODO:
+// X Import marker data into algoia index
+// X Set up Algoia API keys for the index
+// - Replace below code with code to return a promise to load from algolia
+// - look at index settings to set what is returned from Algoli
```
Has this been done yet?

shayantist commented on map/src/algoliaUtil.js:
```diff
@@ -0,0 +1,90 @@
+import { v4 as uuidv4 } from 'uuid';
+
+dotenv.config();
+const algoliaAdminKey = process.env.REACT_APP_ALGOLIA_ADMIN_KEY || 'undefined';
+const algoliaAppId = process.env.REACT_APP_ALGOLIA_APP_ID || 'undefined';
+const client = algoliasearch(algoliaAppId, algoliaAdminKey);
```
Why are these being defined again if they're in [dataDriver.ts](https://github.com/reach4help/reach4help/blob/map-algolia/map/src/data/dataDriver.ts)? What purpose does this file serve?

shayantist reviewed (changes requested):
Really well done, @ethanstrominger and @jensren . The transition from Firestore to Algolia feels almost seamless functionality-wise, while providing huge performance gains. Once we implement full-text search and migrate the entire Firestore db over to Algolia (bonus points for geo-search), then we'll be ready to merge this. 

What purpose do the files under [algolia_scripts](https://github.com/reach4help/reach4help/tree/map-algolia/map/src/algolia-scripts) serve?

I see a lot of changed files under [parse_scripts](https://github.com/reach4help/reach4help/tree/map-algolia/map/src/parse-scripts), which likely overwrites our data work on the map branch. Since this PR is only for changes related to Algolia, better to remove them to prevent conflicts. Check some of my other comments on specific files as well.

ethanstrominger committed:
fix: sycn parse_scripts from map branch to map-algolia

ethanstrominger committed:
fix: add levenary otherwise build err

ssameghini committed:
Set initial location on client geolocation

ethanstrominger committed:
fix: error in dataDriver.ts

ethanstrominger committed:
refactor: remove TODO comments

jensren committed:
refactor: generalize FilterType component

Co-authored-by: Ethan-Strominger <ethanstrominger@users.noreply.github.com>

ethanstrominger committed:
fix: set added marker not visible, review comments

ethanstrominger committed:
feat: map search text

jensren committed:
move: MyLocation before 'Search text' in MapLayout

jensren committed:
feat: add service filter

ssameghini committed:
Position map on users given location on first load

jensren committed:
fix: commit errors

jensren committed:
refactor: incorporate PR comments

Co-authored-by: Ethan-Strominger <ethanstrominger2@gmail.com>

jensren committed:
refactor: filterScreenField prop

Co-authored-by: Ethan-Strominger <ethanstrominger@users.noreply.github.com>

ssameghini committed:
Remove unnecessary condition for localStorage

jensren committed:
fix: cannot find R4HGeoPoint

ethanstrominger committed:
Merge pull request #1401 from reach4help/jen/map-algolia/1271-query-service

Jen/map algolia/1271 query service

ethanstrominger committed:
fix: case issue

ethanstrominger committed:
Update and rename R4hGeoPoint.ts to R4HGeoPoint.ts

ethanstrominger committed:
fix: lint error

ssameghini committed:
Feat: geolocate in users position or over India by default

ssameghini committed:
Feat: collapse search bar achieved, styling pendant

jensren committed:
dev: add more test markers

ssameghini committed:
Feat: collapse search section with language and styling

jensren committed:
fix: remove any type casting

jensren committed:
fix: look up 'any' based on translationKey

jensren committed:
docs: add docs

ethanstrominger committed:
Merge pull request #1406 from reach4help/jen/map-algolia/1271-query-service

Add more test markers, fix some problems, add docs

ssameghini committed:
Merge branch 'map-algolia' of https://github.com/reach4help/reach4help into ssameghini/map-algolia/issue-1405-collapse_search_bar

ethanstrominger committed:
Merge pull request #1404 from reach4help/ssameghini/map-algolia/issue-1194-location_default_on_user

Feat: geolocate on users position in the first visit, or over India by default

ssameghini committed:
Fix: refactor state management

ssameghini committed:
Merge branch 'map-algolia' of https://github.com/reach4help/reach4help into ssameghini/map-algolia/issue-1405-collapse_search_bar

ssameghini committed:
Refactor(geolocation): delete button and extract method

My Location button gets erased and the geolocation function in Map is moved up to a component method.

jensren committed:
refactor: PR #1406 comments

Co-authored-by: Ethan-Strominger <ethanstrominger2@gmail.com>

jensren committed:
feat: multiselect filter

ssameghini committed:
Delete My Location for merging with upstream

ssameghini committed:
Merge branch map-algolia into issue 1405

jensren committed:
refactor: filter lists to sets, remove unneeded prop

ethanstrominger committed:
Merge pull request #1407 from reach4help/ssameghini/map-algolia/issue-1405-collapse_search_bar

Feat: Make the search controls section collapsible (close #1405)

mitchell-verter-zocdoc committed:
default color needed for information and financial markers

jensren committed:
feat: style filter panel

ssameghini committed:
Merge branch 'map-algolia' of https://github.com/reach4help/reach4help into mitchell-sebastian/map/issue-1194-location_default_to_user

jensren committed:
fix: add translations for filter labels

ssameghini committed:
Fix: Geolocate with API by default

ethanstrominger committed:
Merge pull request #1420 from reach4help/mitchell-sebastian/map/issue-1194-location_default_to_user

Fix: Geolocate with an API by default

ethanstrominger committed:
Merge pull request #1418 from reach4help/jen/map-algolia/1371-multi-select

#1371: multi select

ethanstrominger committed:
Merge pull request #1416 from reach4help/fix-color-crash

default color needed for information and financial markers

jensren committed:
fix: selected drop-down options not translating

jensren committed:
fix: scroll panel when too long

ethanstrominger committed:
Merge pull request #1424 from reach4help/jen/map-algolia/1419-translation-bug

fix: selected drop-down options not translating

ethanstrominger committed:
feat: rm dir not part of feature

jensren committed:
update: react-select and @types/react-select

jensren committed:
feat: anchor panel

ethanstrominger committed:
Merge pull request #1482 from reach4help/jen/map-algolia/1421-panel-scroll

Fix: panel scroll when too many options selected
